src/bank.rs

```diff
@@ -309,34 +328,87 @@ impl Module for BankKeeper {
             self.get_balance(&bank_storage, &Addr::unchecked(IBC_LOCK_MODULE_ADDRESS))?;
         let locked_amount = balances.iter().find(|b| b.denom == packet.denom);
 
-        if let Some(locked_amount) = locked_amount {
+        let contract_exec = if let Some(memo) = packet.memo {
+            // We match the memo to the IBC hooks format
+            // If it matches, we create the ibc hook sender. They will be the recipient of the funds and the sender of the contract call
+            let json: Value = serde_json::from_str(&memo)?;
```

Comment:
```suggestion
            let json: serde_json::Result<Value> = serde_json::from_str(&memo);
```
We can't assume any memo is a json